src/lib.rs

```diff
@@ -240,20 +255,21 @@ pub fn load_zones(
 
 // TODO Share with rampfs
 fn points_per_polygon(
-    points: Vec<Point<f64>>,
+    points: Vec<WeightedPoint>,
     polygons: &HashMap<String, MultiPolygon<f64>>,
-) -> BTreeMap<String, Vec<Point<f64>>> {
+) -> BTreeMap<String, Vec<WeightedPoint>> {
     let tree = RTree::bulk_load(points);
 
     let mut output = BTreeMap::new();
     for (key, polygon) in polygons {
         let mut pts_inside = Vec::new();
         let bounds = polygon.bounding_rect().unwrap();
-        let envelope: AABB<Point<f64>> =
-            AABB::from_corners(bounds.min().into(), bounds.max().into());
+        let min = bounds.min();
+        let max = bounds.max();
+        let envelope: AABB<[f64; 2]> = AABB::from_corners([min.x, min.y], [max.x, max.y]);
         for pt in tree.locate_in_envelope(&envelope) {
-            if polygon.contains(pt) {
-                pts_inside.push(*pt);
+            if polygon.contains(&pt.point) {
+                pts_inside.push(pt.clone());
```

Comment:
guess you cannot use it twice hence the clone?

Reply:
`locate_in_envelope` doesn't remove from the rtree, it returns a borrow. Is cheap to copy here